route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -109,13 +97,36 @@ namespace route_following_plugin
             }
             if(identifyLaneChange(following_lanelets, shortest_path[last_lanelet_index + 1].id()))
             {
-                ++last_lanelet_index;
+                //calculate start distance
+                double longl_acceleration=(RouteFollowingPlugin::TWENTY_FIVE_MPH_IN_MS-speed_progress)/LANE_CHANGE_TIME_MAX;
+                double lane_change_dist=(speed_progress*LANE_CHANGE_TIME_MAX)+0.5*(longl_acceleration)*pow(LANE_CHANGE_TIME_MAX,2);
+                double start_dist=end_dist-current_progress-lane_change_dist;
+                double start_time=start_dist/speed_progress;
+                //lane following till start_distance
+                resp.new_plan.maneuvers.push_back(
+                composeManeuverMessage(current_progress, start_dist, 
+                                       speed_progress, speed_progress, 
+                                       shortest_path[last_lanelet_index].id(), ros::Time::now()));
```

Comment:
Looking at how `composeManeuverMessage` works, I think this `ros::Time::Now()` needs to update based on the end time of the previous maneuver. The parameter might be named `current_time` but it's used to set the start time of each maneuver, and currently they all start at the beginning of the maneuver plan, overlapping instead of being in sequence.